gremlin/traversal/metrics.go

```diff
@@ -272,7 +272,25 @@ func (m *MetricsTraversalStep) Count(s ...interface{}) *traversal.GraphTraversal
 	return traversal.NewGraphTraversalValue(m.GraphTraversal, len(m.metrics))
 }
 
+func (m *MetricsTraversalStep) assert() {
+	if m.error != nil {
+		return
+	}
+	if m.GraphTraversal == nil {
+		panic("MetricsTraversalStep.GraphTraversal should not be nil!")
+	}
+}
+
 // NewMetricsTraversalStep creates a new traversal metric step
-func NewMetricsTraversalStep(gt *traversal.GraphTraversal, metrics map[string][]common.Metric, err error) *MetricsTraversalStep {
-	return &MetricsTraversalStep{GraphTraversal: gt, metrics: metrics, error: err}
+func NewMetricsTraversalStep(gt *traversal.GraphTraversal, metrics map[string][]common.Metric) *MetricsTraversalStep {
+	m := &MetricsTraversalStep{GraphTraversal: gt, metrics: metrics}
+	m.assert()
+	return m
+}
+
+// NewMetricsTraversalStepFromError creates a new traversal metric step
+func NewMetricsTraversalStepFromError(err error) *MetricsTraversalStep {
+	m := &MetricsTraversalStep{error: err}
+	m.assert()
```

Comment:
I don't see the value of assert : if there is an error, it does nothing. Otherwise it panics, as it would have done anyway.

Reply:
will remove it